### PR TITLE
Adds sending power monitor events to the jitsi-meet iframe.

### PR DESCRIPTION
**Pull request by damencho**

### CODE REVIEW

File: powermonitor/render.js [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+    }
+}
+
+/**
+ * Initializes the power monitor in the render process of the
+ * window which displays Jitsi Meet.
+ *
+ * @param iframe - the iframe to send events to.
+ */
+module.exports = function setupPowerMonitorRender(iframe) {
```
Comment (hristoterezov) [inline powermonitor/render.js]: I know we require the iframe directly as a parameter for some of our electron APIs but isn't it going to be better to require the api instance and get iframe from it. Looks a little bit more easier for the consumers and also if we ever need something from the api in future, we won't need to change the electron API (this happened to me once in the past).

File: powermonitor/render.js [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+        _channel = postis({
+            window: iframe.contentWindow,
+            windowForEventListening: window,
+            scope: 'jitsi-power-monitor'
+        });
+        _channel.ready(() => {
+            _channel.listen('message', message => {
+                const { name } = message.data;
+                if(name === POWER_MONITOR_MESSAGE_NAME) {
+                    ipcRenderer.send(POWER_MONITOR_QUERIES_CHANNEL, message);
```
Comment (hristoterezov) [inline powermonitor/render.js]: We are sending a message to us here?

File: powermonitor/render.js [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+ * The channel we use to communicate with Jitsi Meet window.
+ */
+let _channel;
+
+/**
+ * The listener for query responses.
+ * @param _ - Not used.
+ * @param response - The response to send.
+ */
+let queriesChannelListener = function (_, response) {
```
Comment (hristoterezov) [inline powermonitor/render.js]: Why this function definition and the next one are different than all other function definitions in the PR?

File: powermonitor/render.js (hunk 1 above)
Comment (damencho) [reply powermonitor/render.js]: Ok, will change it.

File: powermonitor/render.js (hunk 2 above)
Comment (damencho) [reply powermonitor/render.js]: Yep, this sends a message back to the main process.

File: powermonitor/render.js (hunk 3 above)
Comment (damencho) [reply powermonitor/render.js]: Cause these a the listener we want to remove later.

File: powermonitor/render.js [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+    _sendEvent(event);
+};
+
+/**
+ * Sends event to Jitsi Meet.
+ *
+ * @param {Object} event the remote control event.
+ */
+function _sendEvent(event) {
+    const powerMonitorEvent = Object.assign(
```
Comment (hristoterezov) [inline powermonitor/render.js]: Redundant variable.

File: powermonitor/main.js [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+}
+
+/**
+ * Initializes the power monitor functionality in the main electron process.
+ *
+ * @param {BrowserWindow} jitsiMeetWindow - the BrowserWindow object which
+ * displays Jitsi Meet
+ */
+module.exports = function setupPowerMonitorMain(jitsiMeetWindow) {
+    app.isReady() ? _attachEvents(jitsiMeetWindow)
```
Comment (hristoterezov) [inline powermonitor/main.js]: Optional: I'd use the good old `if` here.

File: powermonitor/render.js [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+        method: 'message',
+        params: message
+    });
+}
+
+/**
+ * Disposes the power monitor functionality.
+ */
+function dispose() {
+
```
Comment (hristoterezov) [inline powermonitor/render.js]: Redundant empty line

File: powermonitor/render.js [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+            'unload',
+            dispose
+        );
+        _channel = postis({
+            window: iframe.contentWindow,
+            windowForEventListening: window,
+            scope: 'jitsi-power-monitor'
+        });
+        _channel.ready(() => {
+            _channel.listen('message', message => {
```
Comment (hristoterezov) [inline powermonitor/render.js]: Do we receive such a messages?

Comment (damencho) [reply powermonitor/render.js]: yes

**Pull request merged**